src/sqlfluff/core/linter/linted_file.py

```diff
@@ -0,0 +1,491 @@
+                pre_hint = self.templated_file.templated_str[
+                    patch.templated_slice.start - 10 : patch.templated_slice.start
+                ]
+            else:
+                pre_hint = self.templated_file.templated_str[
+                    : patch.templated_slice.start
+                ]
+            if patch.templated_slice.stop + 10 < len(self.templated_file.templated_str):
+                post_hint = self.templated_file.templated_str[
+                    patch.templated_slice.stop : patch.templated_slice.stop + 10
+                ]
+            else:
+                post_hint = self.templated_file.templated_str[
+                    patch.templated_slice.stop :
+                ]
+            linter_logger.debug(
+                "        Templated Hint: ...%r <> %r...", pre_hint, post_hint
+            )
+
+            # Attempt to convert to source space.
+            try:
+                source_slice = self.templated_file.templated_slice_to_source_slice(
+                    patch.templated_slice,
+                )
+            except ValueError:
+                linter_logger.info(
+                    "      - Skipping. Source space Value Error. i.e. attempted insertion within templated section."
+                )
+                # If we try and slice within a templated section, then we may fail
+                # in which case, we should skip this patch.
+                continue
+
+            # Check for duplicates
+            dedupe_tuple = (source_slice, patch.fixed_raw)
+            if dedupe_tuple in dedupe_buffer:
+                linter_logger.info(
+                    "      - Skipping. Source space Duplicate: %s", dedupe_tuple
+                )
+                continue
+
+            # We now evaluate patches in the source-space for whether they overlap
+            # or disrupt any templated sections.
+            # The intent here is that unless explicitly stated, a fix should never
+            # disrupt a templated section.
+            # NOTE: We rely here on the patches being sorted.
+            # TODO: Implement a mechanism for doing templated section fixes. For
+            # now it's just not allowed.
+
+            # Get the affected raw slices.
+            local_raw_slices = self.templated_file.raw_slices_spanning_source_slice(
+                source_slice
+            )
+            local_type_list = [slc.slice_type for slc in local_raw_slices]
+
+            enriched_patch = EnrichedFixPatch(
+                source_slice=source_slice,
+                templated_slice=patch.templated_slice,
+                patch_type=patch.patch_type,
+                fixed_raw=patch.fixed_raw,
+                templated_str=self.templated_file.templated_str[patch.templated_slice],
+                source_str=self.templated_file.source_str[source_slice],
+            )
+
+            # Deal with the easy case of only literals
+            if set(local_type_list) == {"literal"}:

```

Comment:
Minor performance question (feel free to ignore -- I'm probably being obsessive, ha ha): How big is `local_type_list` likely to be? If it could be more than a few elements in size, we could end up materializing a large-ish temporary collection for not much reason.

I found a couple of web pages with some clever ways to check the same thing more efficiently.
https://stackoverflow.com/questions/3844801/check-if-all-elements-in-a-list-are-identical
https://thispointer.com/python-check-if-all-elements-in-a-list-are-same-or-matches-a-condition/

Reply:
It's almost always a small sample here. I'm going to leave it as is - but it's a good suggestion.